### ISSUE TITLE
Important issues in Binutils and GCC for AT 15.0-1

### ISSUE THREAD

**Issue by tuliom** [advancetoolchain/advance-toolchain#2352]
We'd like to use this issue in order to track fixes for Binutils and GCC for AT 15.0-1.

- [x] [Binutils bug 28192](https://sourceware.org/bugzilla/show_bug.cgi?id=28192)
- [x] https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=b0963c4379becafaebd8e52b0b42698ff151c293
- [x] https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=69feb7601e86274fa9abbfb420b00c8adf947e7b
- [x] https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=6278065af07634278ba30029d92a82b089969baa
- [x] https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=42a9e4e1381ba14d8ef21b331494945c2f51e6be

**Comment by peter-bergner** [comment]
@tuliom: what form do you want this information?  I have some IBM GCC 11 git log commits for those entries I think are important.  Do you just want me to cut/paste that here or just give a synopsis of the entries?

**Comment by tuliom** [comment]
> what form do you want this information?

@peter-bergner Whatever is best for you.  Commit IDs are great! Bugzilla entries too.

> Do you just want me to cut/paste that here or just give a synopsis of the entries?

Whatever is easier for you.  I just want to track if the important fixes have properly landed in AT 15.0-1.

**Comment by peter-bergner** [comment]
It's easiest just to cut/paste git log entries, so if you're fine with that, then that's what I'll do.

**Comment by peter-bergner** [comment]
Some of the entries are pretty large, so I think I will remove the ChangeLog portion of the commit logs and just give you the commit log message minus the ChangeLog entry.

**Comment by peter-bergner** [comment]
```
commit da0a5628d47ee59320dac9787f3232b0911fbc14
Author:     Segher Boessenkool <segher@kernel.crashing.org>
AuthorDate: Wed Sep 8 13:10:30 2021 +0000
Commit:     Segher Boessenkool <segher@kernel.crashing.org>
CommitDate: Tue Sep 21 23:22:44 2021 +0000

    rs6000: Fix ELFv2 r12 use in epilogue
    
    We cannot use r12 here, it is already in use as the GEP (for sibling
    calls).
    
    (cherry picked from commit 86e6268cff328e27ee6f90e2afc35b6f437a25cd)

commit 5210f05c6eba12561914cb0fdf7aec149b28f179
Author:     Segher Boessenkool <segher@kernel.crashing.org>
AuthorDate: Thu Sep 2 16:38:24 2021 +0000
Commit:     Segher Boessenkool <segher@kernel.crashing.org>
CommitDate: Tue Sep 21 23:21:49 2021 +0000

    rs6000: Don't use r12 for CR save on ELFv2 (PR102107)
    
    CR is saved and/or restored on some paths where GPR12 is already live
    since it has a meaning in the calling convention in the ELFv2 ABI.
    
    It is not completely clear to me that we can always use r11 here, but
    it does seem save, there is checking code (to detect conflicts here),
    and it is stage 1.  So here goes.
    
    (cherry picked from commit 2484f7a4b0f52e6ed04754be336f1fa6fde47f6b)

commit b429c81170ccfa49894404524dac12be3704c574
Author:     Pat Haugen <pthaugen@linux.ibm.com>
AuthorDate: Mon Aug 30 10:58:21 2021 -0500
Commit:     Pat Haugen <pthaugen@linux.ibm.com>
CommitDate: Wed Sep 15 18:02:55 2021 -0500

    Enable store fusion on Power10.
    
commit ed9006264f657bf0ec901af9e511c32b9b67926d
Author:     Peter Bergner <bergner@linux.ibm.com>
AuthorDate: Wed Jul 14 18:27:02 2021 -0500
Commit:     Peter Bergner <bergner@linux.ibm.com>
CommitDate: Wed Sep 15 12:18:24 2021 -0500

    rs6000: Generate an lxvp instead of two adjacent lxv instructions
    
    The MMA build built-ins currently use individual lxv instructions to
    load up the registers of a __vector_pair or __vector_quad.  If the
    memory addresses of the built-in operands are to adjacent locations,
    then we can use an lxvp in some cases to load up two registers at once.
    The patch below adds support for checking whether memory addresses are
    adjacent and emitting an lxvp instead of two lxv instructions.
    
    (cherry picked from commit 69feb7601e86274fa9abbfb420b00c8adf947e7b)

commit a87d7fbef55f72781905bffb298aab698fe6ed40
Author:     Xionghu Luo <luoxhu@linux.ibm.com>
AuthorDate: Mon Sep 6 20:22:50 2021 -0500
Commit:     Xionghu Luo <luoxhu@linux.ibm.com>
CommitDate: Tue Sep 14 00:32:56 2021 -0500

    rs6000: Expand fmod and remainder when built with fast-math [PR97142]
    
    fmod/fmodf and remainder/remainderf could be expanded instead of library
    call when fast-math build, which is much faster.
    
    (cherry-picked from 546ecb0054af302acf0839c7f3eb78598f8c0672)

commit 585667735e9fb7a38357a84d4d25206a8ccec576
Author:     Peter Bergner <bergner@linux.ibm.com>
AuthorDate: Thu Aug 19 17:33:29 2021 -0500
Commit:     Peter Bergner <bergner@linux.ibm.com>
CommitDate: Thu Sep 2 14:09:24 2021 -0500

    rs6000: Fix ICE expanding lxvp and stxvp gimple built-ins [PR101849]
    
    PR101849 shows we ICE on a test case when we pass a non __vector_pair *
    pointer to the __builtin_vsx_lxvp and __builtin_vsx_stxvp built-ins
    that is cast to __vector_pair *.  The problem is that when we expand
    the built-in, the cast has already been removed from gimple and we are
    only given the base pointer.  The solution used here (which fixes the ICE)
    is to catch this case and convert the pointer to a __vector_pair * pointer
    when expanding the built-in.
    
    (cherry picked from commit b0963c4379becafaebd8e52b0b42698ff151c293)

commit 5803a97319245cf7942e544a7be7b24821dc7c9e
Author:     Haochen Gui <guihaoc@gcc.gnu.org>
AuthorDate: Fri Jun 4 14:38:53 2021 +0800
Commit:     Segher Boessenkool <segher@kernel.crashing.org>
CommitDate: Mon Aug 30 09:36:57 2021 +0000

    rs6000: Disable mode promotion for pseudos
    
    rs6000 has instructions that can do almost everything 32 bit
    at least as efficiently as corresponding 64 bit things. The
    mode promotion can be defered to when a wide mode is necessary.
    So it helps a lot not promote mode for pseudos. SPECint test
    shows that the overall performance improvement (by geomean) is
    more than 2% with this patch.
    
    testsuite/gcc.target/powerpc/not-promote-mode.c illustrates how
    the patch eliminates the redundant extensions and do further
    optimization by disabling mode promotion for pseduos.
    
    (cherry picked from commit 9080a3bf23297885fdc47221da37a71d6dec93c5)

commit c8ccb1828b75ca78047983c1b43626fec80e7af0
Author:     Alan Modra <amodra@gmail.com>
AuthorDate: Tue Jun 29 13:31:45 2021 +0930
Commit:     Alan Modra <amodra@gmail.com>
CommitDate: Fri Jul 23 08:54:36 2021 +0930

    [POWER10] __morestack calls from pcrel code
    
    Compiling gcc/testsuite/gcc.dg/split-*.c and others with -mcpu=power10
    and linking with a non-pcrel libgcc results in crashes due to the
    power10 pcrel code not having r2 set for the generic-morestack.c
    functions called from __morestack.  There is also a problem when
    non-pcrel code calls a pcrel libgcc.  See the patch comments.
    
    A similar situation theoretically occurs with ELFv1 multi-toc
    executables, when __morestack might be located in a different toc
    group to its caller.  This patch makes no attempt to fix that, since
    the gold linker does not support multi-toc (gold is needed for proper
    support of -fsplit-stack code) nor does gcc emit __morestack calls
    that support multi-toc.
    
    (cherry picked from commit cd6ca96f5d530e4ee07b65ac8b075119ba5bb035)

commit ac0efe3c6fc6231b20ffd684956a4a5c3c54a96b
Author:     Bill Schmidt <wschmidt@linux.ibm.com>
AuthorDate: Mon Jul 19 12:49:17 2021 -0500
Commit:     Bill Schmidt <wschmidt@linux.ibm.com>
CommitDate: Mon Jul 19 12:52:00 2021 -0500

    rs6000: Don't let swaps pass break multiply low-part (PR101129)
    
commit 8ebcd3608584e544ae8e7c422b3f2400758c47f5
Author:     Michael Meissner <meissner@linux.ibm.com>
AuthorDate: Wed Jul 14 13:23:51 2021 -0400
Commit:     Michael Meissner <meissner@linux.ibm.com>
CommitDate: Wed Jul 14 13:24:56 2021 -0400

    Generate 128-bit int divide/modulus on power10.
    
    This patch adds support for the VDIVSQ, VDIVUQ, VMODSQ, and VMODUQ
    instructions to do 128-bit arithmetic.
    
commit ad522a0acc99efd23d7fb412e5a6af7f8cd9d15c
Author:     Peter Bergner <bergner@linux.ibm.com>
AuthorDate: Fri Jul 2 11:27:52 2021 -0500
Commit:     Peter Bergner <bergner@linux.ibm.com>
CommitDate: Wed Jul 7 11:50:49 2021 -0500

    rs6000: Add MMA __builtin_vsx_lxvp and __builtin_vsx_stxvp built-ins
    
    Add support for the __builtin_vsx_lxvp and __builtin_vsx_stxvp built-ins
    to keep in sync with LLVM which added these earlier.
    
    (cherry picked from commit 42a9e4e1381ba14d8ef21b331494945c2f51e6be)

commit 2d09824484a8bf9c6d101c9b8402d233e32b96c0
Author:     Michael Meissner <meissner@linux.ibm.com>
AuthorDate: Wed Jul 7 12:01:37 2021 -0400
Commit:     Michael Meissner <meissner@linux.ibm.com>
CommitDate: Wed Jul 7 12:08:27 2021 -0400

    Backport IEEE 128-bit min/max code.
    
    This patch back ports 2 IEEE 128-bit min/max patches from the trunk to GCC
    11.  One patch is from June 30th, 2021, and the other from July 1st, 2021.
    
commit ea2f428bfef43d286e0aa789eccd120d9b829a1e
Author:     Aaron Sawdey <acsawdey@linux.ibm.com>
AuthorDate: Tue Jun 22 16:02:15 2021 -0500
Commit:     Aaron Sawdey <acsawdey@linux.ibm.com>
CommitDate: Wed Jun 23 15:08:48 2021 -0500

    Do not enable pcrel-opt by default
    
    Backported from trunk.
    
    SPEC2017 testing on p10 shows that this optimization does not have a
    positive impact on performance. So we are no longer going to enable it
    by default. The test cases for it needed to be updated so they always
    enable it to test it.
    
    (cherry picked from commit b7574ca6edb5ac7fab741f46d4f50ac4294c8180)

commit 8d21764dcbe15f1faec5804e3df4179ef9ea2089
Author:     Michael Meissner <meissner@linux.ibm.com>
AuthorDate: Wed Jun 23 15:00:16 2021 -0400
Commit:     Michael Meissner <meissner@linux.ibm.com>
CommitDate: Wed Jun 23 15:02:39 2021 -0400

    Backport patch from master branch.
    
    Add IEEE 128-bit min/max support on PowerPC.
    
    This patch adds the support for the IEEE 128-bit floating point C minimum and
    maximum instructions.  The next patch will add the support for using the
    compare and set mask instruction to implement conditional moves.
    
    This patch does not try to re-use the code used for SF/DF min/max
    support.  It defines a separate insn for the IEEE 128-bit support.  It
    uses the code iterator <minmax> to simplify adding both operations.
    
    GCC will not convert ternary operations into using min/max instructions
    provided in this patch unless the user uses -Ofast.  The next patch that adds
    conditional move instructions will enable the ternary conversion in many cases.
    
commit 88b66b376844fe7c537ab229250a41a54e706eaf
Author:     Carl Love <cel@us.ibm.com>
AuthorDate: Tue Jun 15 11:37:31 2021 -0500
Commit:     Carl Love <cel@us.ibm.com>
CommitDate: Mon Jun 21 17:33:34 2021 -0500

    RS6000 Add 128-bit Binary Integer sign extend operations
    
    This patch adds the 128-bit sign extension instruction support and
    corresponding builtin support.
    
    RS6000 Add 128-bit Binary Integer sign extend operations
    
commit e7147f01afc3d90d8e96cf4f2899c0f986177674
Author:     Carl Love <cel@us.ibm.com>
AuthorDate: Wed Apr 21 18:07:39 2021 -0400
Commit:     Carl Love <cel@us.ibm.com>
CommitDate: Mon Jun 21 17:33:31 2021 -0500

    Conversions between 128-bit integer and floating point values.
    
    The files fixkfti-sw.c and fixunskfti-sw.c are renamed versions of
    fixkfti.c and fixunskfti.c respectively to do the conversions in software.
    The function names in the files were updated with the rename as well as
    some white spaces fixes. The file float128-p10.c contains the functions
    for using the ISA 3.1 hardware instructions to perform the conversions.
    
commit ac9afc035a9b53af3721623ab3062e1573f8662f
Author:     Carl Love <cel@us.ibm.com>
AuthorDate: Wed Sep 16 16:05:49 2020 -0500
Commit:     Carl Love <cel@us.ibm.com>
CommitDate: Mon Jun 21 17:33:29 2021 -0500

    rs6000, Add test 128-bit shifts for just the int128 type.
    
    This patch also renames and moves the VSX_TI iterator from vsx.md to
    VEC_TI in vector.md.  The uses of VEC_TI are also updated.
    
commit b715c100ba06f273c95f7564e40eaf0409690542
Author:     Carl Love <cel@us.ibm.com>
AuthorDate: Tue Nov 24 18:18:05 2020 -0600
Commit:     Carl Love <cel@us.ibm.com>
CommitDate: Mon Jun 21 17:33:26 2021 -0500

    Add 128-bit int to 128-bit DFP (floattitd2) and 128-bit DFP to 128-bit int (fixtdti2) support
    
commit 2937d4d2919fb63120f2efed25fa4c2ec9c64079
Author:     Carl Love <cel@us.ibm.com>
AuthorDate: Tue Jun 15 11:30:18 2021 -0500
Commit:     Carl Love <cel@us.ibm.com>
CommitDate: Mon Jun 21 17:33:07 2021 -0500

    RS6000 add 128-bit Integer Operations part 1
    
commit 04a6793eec7f3482f3fb1273a725b3001865a685
Author:     Peter Bergner <bergner@linux.ibm.com>
AuthorDate: Mon Jun 14 16:55:18 2021 -0500
Commit:     Peter Bergner <bergner@linux.ibm.com>
CommitDate: Thu Jun 17 17:04:11 2021 -0500

    rs6000: MMA builtin usage ICEs when used in a #pragma omp parallel and using -fopenmp [PR100777]
    
    Using an MMA builtin within an openmp parallel code block, leads to an SSA
    verification ICE on the temporaries we create while expanding the MMA builtins
    at gimple time.  The solution is to use create_tmp_reg_or_ssa_name(), which
    knows when to create either an SSA or register temporary.
    
    (cherry picked from commit 20073534c0ccca0a4e079c053ee0874af10b2ea0)

commit de6f0c41e9da2195eebe6c1f7c459615c8f1ad58
Author:     Peter Bergner <bergner@linux.ibm.com>
AuthorDate: Thu Jun 10 13:54:12 2021 -0500
Commit:     Peter Bergner <bergner@linux.ibm.com>
CommitDate: Thu Jun 17 17:04:11 2021 -0500

    rs6000: Add new __builtin_vsx_build_pair and __builtin_mma_build_acc built-ins
    
    The __builtin_vsx_assemble_pair and __builtin_mma_assemble_acc built-ins
    currently assign their first source operand to the first VSX register
    in a pair/quad, their second operand to the second register in a pair/quad, etc.
    This is not endian friendly and forces the user to generate different calls
    depending on endianness.  In agreement with the POWER LLVM team, we've
    decided to lightly deprecate the assemble built-ins and replace them with
    "build" built-ins that automatically handle endianness so the same built-in
    call and be used for both little-endian and big-endian compiles.  We are not
    removing the assemble built-ins, since there is code in the wild that use
    them, but we are removing their documentation to encourage the use of the
    new "build" variants.
    
    (cherry picked from commit 00d07ec6e12451acc7a290cd93be03bed50cb666)

commit fb7327aa4ddb2fdb846c9b50b56f8c87dc37c8a2
Author:     Aaron Sawdey <acsawdey@linux.ibm.com>
AuthorDate: Tue Mar 2 18:06:37 2021 -0600
Commit:     Aaron Sawdey <acsawdey@linux.ibm.com>
CommitDate: Thu Jun 17 10:59:28 2021 -0500

    Backported from trunk:
    
    Fusion patterns for add-logical/logical-add
    
    This patch modifies the function in genfusion.pl for generating
    the logical-logical patterns so that it can also generate the
    add-logical and logical-add patterns which are very similar.
    
    Also backported from trunk and combined with the add-logical patch
    because that revealed problems on gcc-11.
    
    Add needed earlyclobber to fusion patterns
    
    The add-logical and add-add fusion patterns all have constraint
    alternatives "=0,1,&r,r" for the output (3). The inputs 0 and 1
    are used in the first fusion instruction and then either may be
    reused as a temp for the output of the first insn which is
    input to the second. However, if input 2 is the same as 0 or 1,
    it gets clobbered unexpectedly. So the first 2 alts need to be
    "=&0,&1,&r,r" instead to indicate that in alts 0 and 1, the
    register used for 3 is earlyclobber, hence can't be the same as
    input 2.
    
    This was actually encountered in the backport of the add-logical
    fusion patch to gcc-11. Some code in go hit this case:
    
       <runtime.fillAligned+520>:        andc r30,r30,r9
    r30 now (~(x|((x&c)+c)))&(~c) --> this is new x
       <runtime.fillAligned+524>:        b <runtime.fillAligned+288>
       <runtime.fillAligned+288>:        addi r31,r31,-1
    r31 now m-1
       <runtime.fillAligned+292>:        srd r31,r30,r31
    r31 now x>>(m-1)
       <runtime.fillAligned+296>:        subf r30,r31,r30
    r30 now x-(x>>(m-1))
       <runtime.fillAligned+300>:        or r30,r30,r30   # mdoom
    nop
       <runtime.fillAligned+304>:        not     r3,r30
    r3 now ~(x-(x>>(m-1))) -- WHOOPS
    
    The or r30,r30,r30 was meant to be or-ing in the earlier value
    of r30 which was overwritten by the output of the subf.
    
commit a677a2d513aa29f5963dd82ee50b83fe7d3d4227
Author:     Aaron Sawdey <acsawdey@linux.ibm.com>
AuthorDate: Mon Jan 25 21:11:52 2021 -0600
Commit:     Aaron Sawdey <acsawdey@linux.ibm.com>
CommitDate: Mon Jun 14 17:27:44 2021 -0500

    combine patterns for add-add fusion
    
    This patch adds a function to genfusion.pl to add a couple
    more patterns so combine can do fusion of pairs of add and
    vaddudm instructions.
    
commit c3060361d66fb818d2cbcac265884074cfde4534
Author:     Aaron Sawdey <acsawdey@linux.ibm.com>
AuthorDate: Tue Mar 2 17:50:52 2021 -0600
Commit:     Aaron Sawdey <acsawdey@linux.ibm.com>
CommitDate: Thu Jun 10 20:54:21 2021 -0500

    Add insn types for fusion pairs
    
    This adds new values for insn attr type for p10 fusion. The genfusion.pl
    script is modified to use them, and fusion.md regenerated to capture
    the new patterns. There are also some formatting only changes to
    fusion.md that apparently weren't captured after a previous commit
    of genfusion.pl.
    
commit 426241cd9eac844b7f4f8741a8f540b4a2b5f4c7
Author:     Bill Schmidt <wschmidt@linux.ibm.com>
AuthorDate: Thu May 13 13:10:10 2021 -0500
Commit:     Bill Schmidt <wschmidt@linux.ibm.com>
CommitDate: Wed May 19 15:19:57 2021 -0500

    rs6000: Conditionally define __ROP_PROTECT__
    
commit 7db558884730cfe7a874e9316a5a66987b020964
Author:     Bill Schmidt <wschmidt@linux.ibm.com>
AuthorDate: Fri May 14 14:25:25 2021 -0500
Commit:     Bill Schmidt <wschmidt@linux.ibm.com>
CommitDate: Wed May 19 15:19:49 2021 -0500

    rs6000: Emit ROP-mitigation instructions in prologue and epilogue
    
commit 4ffe506d025e953a0ab3ef0392867d1a46bb9ff5
Author:     Bill Schmidt <wschmidt@linux.ibm.com>
AuthorDate: Fri May 14 14:15:55 2021 -0500
Commit:     Bill Schmidt <wschmidt@linux.ibm.com>
CommitDate: Wed May 19 15:19:42 2021 -0500

    rs6000: Add -mrop-protect and -mprivileged flags
    
commit 9e80a135fffa5c1b36b6007e7e57d303535fbe84
Author:     Michael Meissner <meissner@linux.ibm.com>
AuthorDate: Tue Apr 27 10:52:57 2021 -0400
Commit:     Michael Meissner <meissner@linux.ibm.com>
CommitDate: Tue Apr 27 10:52:57 2021 -0400

    [PATCH] Backport fix for PR target/98952
    
    The test in the PowerPC 32-bit trampoline support is backwards.  It aborts
    if the trampoline size is greater than the expected size.  It should abort
    when the trampoline size is less than the expected size.  I fixed the test
    so the operands are reversed.  I then folded the load immediate into the
    compare instruction.
    
    I verified this by creating a 32-bit trampoline program and manually
    changing the size of the trampoline to be 48 instead of 40.  The program
    aborted with the larger size.  I updated this code and ran the test again
    and it passed.
    
    I added a test case that runs on PowerPC 32-bit Linux systems and it calls
    the __trampoline_setup function with a larger buffer size than the
    compiler uses.  The test is not run on 64-bit systems, since the function
    __trampoline_setup is not called.  I also limited the test to just Linux
    systems, in case trampolines are handled differently in other systems.
    
commit 8bd3b0c11cb08f63d2ca41a21596d16635626661
Author:     Segher Boessenkool <segher@kernel.crashing.org>
AuthorDate: Tue Apr 20 12:00:50 2021 +0000
Commit:     Segher Boessenkool <segher@kernel.crashing.org>
CommitDate: Tue Apr 20 13:55:04 2021 +0000

    rs6000: Fix cpu selection w/ isel (PR100108)
    
    There are various non-IBM CPUs with isel as well, so it is easiest if we
    just don't consider that flag here (it is not needed).
    
    (cherry picked from commit 6156df483fa50a08f561b6c248819f2992aa380d)
```

**Comment by peter-bergner** [comment]
I've left off commits from us that only updated test cases or are only AIX related.

**Comment by peter-bergner** [comment]
Remind me which binutils release we are uysing and I'll try and come up with a similar list for that.

**Comment by tuliom** [comment]
> Remind me which binutils release we are uysing and I'll try and come up with a similar list for that.

That's Binutils 2.37.

**Comment by tuliom** [comment]
The following GCC commits have been merged via commit 58d24bbe51778e0f6775fe04c7aa59f54afa61c2.

```
commit da0a5628d47ee59320dac9787f3232b0911fbc14
commit 5210f05c6eba12561914cb0fdf7aec149b28f179
commit b429c81170ccfa49894404524dac12be3704c574
commit ed9006264f657bf0ec901af9e511c32b9b67926d
commit a87d7fbef55f72781905bffb298aab698fe6ed40
commit 585667735e9fb7a38357a84d4d25206a8ccec576
commit 5803a97319245cf7942e544a7be7b24821dc7c9e
commit c8ccb1828b75ca78047983c1b43626fec80e7af0
```

Meanwhile, the following commits have been merged previously with commit 05c555c71ebdb5f6a68ed38a10ca0dd412ca6df4 and should be available on AT 15.0-0:

```
Merge: 96028ae9f96a46372b118ced2530581676f61a7e

commit ac0efe3c6fc6231b20ffd684956a4a5c3c54a96b
commit 8ebcd3608584e544ae8e7c422b3f2400758c47f5
commit ad522a0acc99efd23d7fb412e5a6af7f8cd9d15c
commit 2d09824484a8bf9c6d101c9b8402d233e32b96c0
commit ea2f428bfef43d286e0aa789eccd120d9b829a1e
commit 8d21764dcbe15f1faec5804e3df4179ef9ea2089
commit 88b66b376844fe7c537ab229250a41a54e706eaf
commit e7147f01afc3d90d8e96cf4f2899c0f986177674
commit ac9afc035a9b53af3721623ab3062e1573f8662f
commit b715c100ba06f273c95f7564e40eaf0409690542
commit 2937d4d2919fb63120f2efed25fa4c2ec9c64079
commit 04a6793eec7f3482f3fb1273a725b3001865a685
commit de6f0c41e9da2195eebe6c1f7c459615c8f1ad58
commit fb7327aa4ddb2fdb846c9b50b56f8c87dc37c8a2
commit a677a2d513aa29f5963dd82ee50b83fe7d3d4227
commit c3060361d66fb818d2cbcac265884074cfde4534


Merge: f26b6020ce7d31263616bbc7dab54cdcbe2ba575

commit 426241cd9eac844b7f4f8741a8f540b4a2b5f4c7
commit 7db558884730cfe7a874e9316a5a66987b020964
commit 4ffe506d025e953a0ab3ef0392867d1a46bb9ff5
commit 9e80a135fffa5c1b36b6007e7e57d303535fbe84
commit 8bd3b0c11cb08f63d2ca41a21596d16635626661
```

So, I think we already have all the fixes listed in [this comment](https://github.com/advancetoolchain/advance-toolchain/issues/2352#issuecomment-926951931).
